include/linux/blkdev_compat.h

```diff
@@ -311,10 +311,14 @@ bio_set_flags_failfast(struct block_device *bdev, int *flags)
  * the block layers responsibility to choose the correct way to
  * implement these semantics.
  */
-#ifdef WRITE_FLUSH_FUA
+#if defined(WRITE_FLUSH_FUA)
 #define	VDEV_WRITE_FLUSH_FUA		WRITE_FLUSH_FUA
-#else
+#elif defined(WRITE_BARRIER)
 #define	VDEV_WRITE_FLUSH_FUA		WRITE_BARRIER
+#elif defined(HAVE_REQ_FUA)
+#define	VDEV_WRITE_FLUSH_FUA		REQ_FUA
```

Comment:
You'll want to order the WRITE_BARRIER check before the REQ_FLUSH check since both are defined in the 2.6.32 era kernels.  Just like `bio_is_flush()` does, and is described in the comment.  You should also be able to leverage the existing configure checks.